numba/targets/setobj.py

```diff
@@ -1007,11 +1016,13 @@ def _allocate_payload(self, nentries, realloc=False):
         with builder.if_then(builder.load(ok), likely=True):
             if realloc:
                 meminfo = self._set.meminfo
-                ptr = context.nrt.meminfo_varsize_alloc(builder, meminfo,
+                ptr = context.nrt.meminfo_varsize_realloc(builder, meminfo,
```

Comment:
this line is causing the problem

Reply:
changing it to regular alloc doesn't seem to help.

Reply:
hmmm, it leaked a load but it didn't segfault for me.